packages/asset-value/src/assetValue.ts

```diff
@@ -0,0 +1,212 @@
+    })
+  }
+
+  public negated(): AssetValue {
+    const value = bnOrZero(this.state.value).negated().toFixed()
+    return new AssetValue({
+      value,
+      assetId: this.state.assetId,
+      precision: this.state.precision,
+      format: AssetValueFormat.BASE_UNIT,
+    })
+  }
+
+  public isGreaterThan(term: AssetValue): boolean {
+    this.assertIsSameAsset(term, 'compare')
+    return bnOrZero(this.state.value).gt(bnOrZero(term.toBaseUnit()))
+  }
+
+  public isGreaterThanOrEqualTo(term: AssetValue): boolean {
+    this.assertIsSameAsset(term, 'compare')
+    return bnOrZero(this.state.value).gte(bnOrZero(term.toBaseUnit()))
+  }
+
+  public isLessThan(term: AssetValue): boolean {
+    this.assertIsSameAsset(term, 'compare')
+    return bnOrZero(this.state.value).lt(bnOrZero(term.toBaseUnit()))
+  }
+
+  public isLessThanOrEqualTo(term: AssetValue): boolean {
+    this.assertIsSameAsset(term, 'compare')
+    return bnOrZero(this.state.value).lte(bnOrZero(term.toBaseUnit()))
+  }
+
+  public isEqualTo(term: AssetValue): boolean {
+    this.assertIsSameAsset(term, 'compare')
+    return this.state.value === term.toBaseUnit()
+  }
+
+  public isNegative(): boolean {
+    return bnOrZero(this.state.value).isNegative()
+  }
+
+  public isZero(): boolean {
+    return bnOrZero(this.state.value).isZero()
+  }
+
+  public toBaseUnit(): string {
+    return this.state.value
+  }
+
+  public toPrecision(decimals?: number | string): string {
+    const _decimals = Number(decimals ?? Math.min(this.state.precision, DEFAULT_FORMAT_DECIMALS))
+    return bnOrZero(this.state.value)
+      .dividedBy(bn(10).pow(bnOrZero(this.state.precision)))
+      .toFixed(_decimals)
+  }
+
+  public toFixed(decimalPlaces?: number, roundingMode?: BigNumber.RoundingMode): string {
+    if (decimalPlaces) {
+      return bnOrZero(this.toPrecision()).toFixed(decimalPlaces, roundingMode)
+    }
+    return bnOrZero(this.toPrecision()).toFixed()
+  }
+
+  public toSerialized(): SerializedAssetValue {
+    const serializedState = JSON.stringify({
+      a: this.state.assetId,
+      p: this.state.precision,
+      v: this.state.value,
+    })
+    const hash = Md5.hashStr(serializedState).slice(0, CHECKSUM_LENGTH)
+    return `${serializedState}|${hash}` as SerializedAssetValue
+  }
+
+  public mult = this.multipliedBy.bind(this)
+  public div = this.dividedBy.bind(this)
+  public gte = this.isGreaterThanOrEqualTo.bind(this)
+  public lt = this.isLessThan.bind(this)
+  public lte = this.isLessThanOrEqualTo.bind(this)
+  public eq = this.isEqualTo.bind(this)
+  public base = this.toBaseUnit.bind(this)
+  public prec = this.toPrecision.bind(this)
+
+  private loadSerializedAssetValue(value: SerializedAssetValue) {
+    const [serializedState, hash] = value.split('|')
+    if (!(serializedState && hash)) {
+      throw new Error('Cannot initialize AssetValue from improperly-formatted SerializedAssetValue')
+    }
+    const parsedState = JSON.parse(serializedState)
```

Comment:
I don't recall the thrown error from `JSON.parse`, but it can throw if you want to throw a custom error